include/zenoh-pico/collections/pointer.h

```diff
@@ -19,19 +19,25 @@
 #include <stdint.h>
 
 #if ZENOH_C_STANDARD != 99
+#ifndef __cplusplus
 #include <stdatomic.h>
+#else
+#include <atomic>
+#define _Atomic(X) std::atomic<X>
+using namespace std;
```

Comment:
That doesn't seem to be the right fix, at least for Zephyr because it doesn't have `std::atomic` and the whole of `std`.

Here is the compile error:

```
In file included from ../../lib/zenoh-pico/include/zenoh-pico/session/session.h:22,
                 from ../../lib/zenoh-pico/include/zenoh-pico/net/session.h:19,
                 from ../../lib/zenoh-pico/include/zenoh-pico/net/publish.h:18,
                 from ../../lib/zenoh-pico/include/zenoh-pico/api/types.h:17,
                 from ../../lib/zenoh-pico/include/zenoh-pico/api/primitives.h:21,
                 from ../../lib/zenoh-pico/include/zenoh-pico/api/macros.h:17,
                 from ../../lib/zenoh-pico/include/zenoh-pico.h:28,
                 from ../../src/main.cpp:24:
../../lib/zenoh-pico/include/zenoh-pico/collections/pointer.h:25:10: fatal error: atomic: No such file or directory
   25 | #include <atomic>
      |          ^~~~~~~~
compilation terminated.
```

Reply:
I am assuming you set the following configs in your `prf.conf`
```
CONFIG_CPP=y
CONFIG_CPP_MAIN=y
```

if that is the case you might want to enable C++ standard library by setting [CONFIG_LIB_CPLUSPLUS](https://docs.zephyrproject.org/3.1.0/kconfig.html#CONFIG_LIB_CPLUSPLUS) in the application configuration file.

Although this solves that particular include error, it dump many other that I am now tackling it.

Reply:
And the latest commit should fix the remaining compilation issues.
If you want to give it a try again @dcorbeil

Reply:
Oh no my bad. Turning on `CONFIG_LIB_CPLUSPLUS` fixes it. No problems to report. Thanks